### PR TITLE
Migrate Login endpoint to GQL

### PR DESCRIPTION
**Pull request by lynn-kim**
## Notion ticket link
[Migrate login endpoint to GraphQL](https://www.notion.so/uwblueprintexecs/Migrate-login-endpoint-to-GraphQL-9aee1f27206a48f1ba5b3dcd36512646)

## Implementation description
* implemented the resolver for the login GQL mutation 
* updated the frontend to use the GQL mutation instead of the REST API

## Steps to test
Either through http://localhost:5000/graphql or the frontend,
1. Log in with the planet read admin/user credentials
2. Log in with invalid credentials [should return an error]

## What should reviewers focus on?
* `backend/python/app/graphql/mutations/auth_mutation.py` 
* `frontend/src/components/auth/Login.tsx` (mostly copied from starter-code)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/graphql/mutations/auth_mutation.py [hunk 1]
```diff
@@ -42,3 +44,31 @@ def mutate(root, info, email):
...
+    id = graphene.Int()
+    first_name = graphene.String(required=True)
+    last_name = graphene.String(required=True)
+    role = graphene.Field(RoleEnum, required=True)
+    email = graphene.String(required=True)
+
+    def mutate(root, info, email, password):
+        try:
+            auth_dto = auth_service.generate_token(email=email, password=password)
+            # TODO: is there a syntactically nicer way to destructure this?
```
Comment (gaoxk) [inline backend/python/app/graphql/mutations/auth_mutation.py:64]: Something like the operator module looks interesting:
```
from operator import attrgetter
id, email, gender, username = attrgetter('id', 'email', 'gender', 'username')(current_user)
print(id, email, gender, username)
```
https://stackoverflow.com/questions/54785148/destructuring-dicts-and-objects-in-python

You could also add a deconstruct function to AuthDTO if you'd like (see first stack overflow response)

File: frontend/src/APIClients/AuthAPIClient.ts [hunk 2]
```diff
@@ -6,21 +11,35 @@ import {
   setLocalStorageObjProperty,
 } from "../utils/LocalStorageUtils";
 
+type LoginFunction = (
```
Comment (gaoxk) [inline frontend/src/APIClients/AuthAPIClient.ts:14]: oof. Definitely, a moment where typescript feels like a mega pain in the butt. 
Played around on your branch and found that this is more concise and worked:
```
type LoginFunction = (
  options: MutationFunctionOptions<
    { login: AuthenticatedUser },
    OperationVariables
  >,
) => Promise<FetchResult<{ login: AuthenticatedUser }>>;
```

**Review by gaoxk — CHANGES_REQUESTED** [review]
looking good so far! 🚀

File: frontend/src/APIClients/AuthAPIClient.ts (hunk 2 above)
Comment (lynn-kim) [reply frontend/src/APIClients/AuthAPIClient.ts:14]: yeah, that looks better! I got this straight from starter-code so might be worth noting the change for the original repo?

Comment (gaoxk) [reply frontend/src/APIClients/AuthAPIClient.ts:14]: ah good to know! I'll mention it to the internal tools team

**Review by gaoxk — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**